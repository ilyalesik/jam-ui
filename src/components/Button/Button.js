import React, { useContext } from "react";
import styled, { css, style } from "reshadow";
import { ThemeContext } from "../../utils/ThemeProvider";

export const Button = ({ children, styles, ...props }) => {
    const { color = "#4ba82e" } = useContext(ThemeContext);
    return styled(styles)`
        button {
            display: flex;
            align-items: center;
            justify-content: center;

            width: 225px;
            height: 58px;
            border: none;

            border-radius: 2px;
            background-color: ${color};
        }
        span {
            font-family: Helvetica, sans-serif;
            font-size: 20px;
            line-height: 1.4;
            color: #ffffff;
            text-transform: capitalize;
        }
    `(
        <root {...props}>
            <button>
                <span>{children}</span>
            </button>
        </root>
    );
};

export default Button;
