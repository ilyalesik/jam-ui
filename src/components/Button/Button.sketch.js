// @flow
import * as React from 'react';
import { render, Artboard, Text, View, Image, makeSymbol } from 'react-sketchapp';

export const Button = () => (
    <View name="Button" style={{ width: 100, height: 100, backgroundColor: 'green' }}>
        <Text name="Button Text">Red Square</Text>
    </View>
);

export const ButtonSymbol = makeSymbol(Button, 'button/default');
