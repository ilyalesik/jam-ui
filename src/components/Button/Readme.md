Basic button:

```jsx
<Button>Push Me</Button>
```
