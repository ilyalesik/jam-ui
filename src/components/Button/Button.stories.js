// @flow
import React from "react";
import { storiesOf } from "@storybook/react";
import Button from "./Button";
import { ThemeProvider } from "../../utils/ThemeProvider";
import styled, { css, style } from "reshadow";

storiesOf("Button", module)
    .add("Button", () => <Button>Push Me</Button>)
    .add("Button styled", () => {
        const StyledButton = ({ ...props }) =>
            styled`
  button {
    width: 400px;
  }
`(<Button {...props} {...styled} />);
        return <StyledButton>Push Me</StyledButton>;
    });

storiesOf("Button", module)
    .addDecorator(story => <ThemeProvider options={{ primaryColor: "#ff0000" }}>{story()}</ThemeProvider>)
    .add("Button themed", () => <Button>Push Me</Button>);
