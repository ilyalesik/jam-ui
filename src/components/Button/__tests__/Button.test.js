import React from "react";
import renderer from "react-test-renderer";
import Button from "../Button";
import {getStyles} from 'reshadow'

describe("<Button />", () => {

    it("Button renders correctly", () => {
        const tree = renderer
        .create(
            <Button>Push Me</Button>
        )
        .toJSON();
        expect(tree).toMatchSnapshot();
        expect(getStyles().css).toMatchSnapshot();
    });
});
