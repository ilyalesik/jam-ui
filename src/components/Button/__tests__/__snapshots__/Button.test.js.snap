// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Button renders correctly 1`] = `
<button
  className="___button_xh087_1"
  style={
    Object {
      "--c8_0": "#4ba82e",
    }
  }
>
  <span
    className="___span_xh087_1"
  >
    Push Me
  </span>
</button>
`;

exports[`<Button /> Button renders correctly 2`] = `
"<style type=\\"text/css\\" id=\\"reshadow-2023947909\\">.___button_xh087_1 {
    display: flex;
    align-items: center;
    justify-content: center;

    width: 225px;
    height: 58px;
    border: none;

    border-radius: 2px;
    background-color: var(--c8_0);
}
.___span_xh087_1 {
    font-family: Helvetica, sans-serif;
    font-size: 20px;
    line-height: 1.4;
    color: #ffffff;
    text-transform: capitalize;
}</style>"
`;
